web/source/kmwdom.ts

```diff
@@ -1495,226 +1495,242 @@ namespace com.keyman {
 
         // Otherwise, assume relative to source path
         return this.keyman.srcPath+p;
-      }            
-      
-      // Explicit (user-defined) parameter initialization       
-      opt=this.keyman.options;
-      if(typeof(arg) == 'object' && arg !== null)
-      {
-        for(p in opt)
-        { 
-          if(arg.hasOwnProperty(p)) opt[p] = arg[p];
-        }
-      }
-    
-      // Get default paths and device options
-      if(opt['root'] != '') {
-        this.keyman.rootPath = fixPath(opt['root']); 
       }
 
-      // Keyboards and fonts are located with respect to the server root by default          
-      //if(opt['keyboards'] == '') opt['keyboards'] = keymanweb.rootPath+'keyboard/';
-      //if(opt['fonts'] == '') opt['fonts'] = keymanweb.rootPath+'font/';
-    
-      // Resources are located with respect to the engine by default 
-      if(opt['resources'] == '') {
-        opt['resources'] = this.keyman.srcPath;
-      }
-    
-      // Convert resource, keyboard and font paths to absolute URLs
-      opt['resources'] = fixPath(opt['resources']);
-      opt['keyboards'] = fixPath(opt['keyboards']);
-      opt['fonts'] = fixPath(opt['fonts']);    
-
-      // Set element attachment type    
-      if(opt['attachType'] == '') {
-        opt['attachType'] = 'auto';
-      }
+      var success = true;
+      var error;
+      
+      try {
```

Comment:
To verify `kmwdom.ts`, please use the `?w=1` URL parameter - the vast majority of its changes are whitespacing due to try-catch scoping.